src/test/groovy/com/gooddata/executeafm/afm/PopMeasureDefinitionTest.groovy

```diff
@@ -17,39 +17,66 @@ import static spock.util.matcher.HamcrestSupport.that
 
 class PopMeasureDefinitionTest extends Specification {
 
-    private static final String POP_MEASURE_DEFINITION_JSON = 'executeafm/afm/popMeasureDefinition.json'
+    private static final String POP_MEASURE_JSON = 'executeafm/afm/popMeasureDefinition.json'
+    private static final String POP_MEASURE_WITH_OFFSET_JSON = 'executeafm/afm/popMeasureDefinitionWithOffset.json'
+
 
     def "should serialize"() {
         expect:
-        that new PopMeasureDefinition('mId', new UriObjQualifier('/gdc/md/projectId/obj/1')),
-                jsonEquals(resource(POP_MEASURE_DEFINITION_JSON))
+        that new PopMeasureDefinition('mId', new UriObjQualifier('/gdc/md/projectId/obj/1'), offset),
+                jsonEquals(resource(expectedJsonMeasureDefinition))
+
+        where:
+        offset | expectedJsonMeasureDefinition
+        null   | POP_MEASURE_JSON
+        -2     | POP_MEASURE_WITH_OFFSET_JSON
     }
 
     def "should deserialize"() {
         when:
-        PopMeasureDefinition measure = readObjectFromResource("/$POP_MEASURE_DEFINITION_JSON", PopMeasureDefinition)
+        PopMeasureDefinition measure = readObjectFromResource("/$jsonMeasureDefinition", PopMeasureDefinition)
 
         then:
         measure.measureIdentifier == 'mId'
-        (measure.popAttribute  as UriObjQualifier).uri == '/gdc/md/projectId/obj/1'
+        (measure.popAttribute as UriObjQualifier).uri == '/gdc/md/projectId/obj/1'
         measure.isAdHoc()
         measure.toString()
+        measure.offset == expectedOffset
+
+        where:
+        jsonMeasureDefinition        | expectedOffset
+        POP_MEASURE_JSON             | null
+        POP_MEASURE_WITH_OFFSET_JSON | -2
     }
 
     def "should copy"() {
         when:
-        def measure = new PopMeasureDefinition("mid", new IdentifierObjQualifier("id"))
+        def measure = new PopMeasureDefinition("mId", new IdentifierObjQualifier("id"), inputOffset)
         def copy = measure.withObjUriQualifier(new UriObjQualifier("uri"))
 
         then:
+        copy.measureIdentifier == 'mId'
         copy.objQualifier.uri == 'uri'
+        copy.offset == expectedOffset
+
+        where:
+        inputOffset | expectedOffset
+        null        | null
+        -2          | -2
     }
 
     def "test serializable"() {
-        PopMeasureDefinition measureDefinition = readObjectFromResource("/$POP_MEASURE_DEFINITION_JSON", PopMeasureDefinition)
+        when:
+        PopMeasureDefinition measureDefinition = readObjectFromResource("/$jsonMeasureDefinition", PopMeasureDefinition)
         PopMeasureDefinition deserialized = SerializationUtils.roundtrip(measureDefinition)
 
-        expect:
+        then:
         that deserialized, jsonEquals(measureDefinition)
+
+        where:
+        jsonMeasureDefinition        | _
```

Comment:
better use `jsonMeasureDefinitin << [...]` - no need for the table here

Reply:
Fixed.